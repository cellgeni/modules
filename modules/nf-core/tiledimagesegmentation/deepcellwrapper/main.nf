process TILEDIMAGESEGMENTATION_DEEPCELLWRAPPER {
    tag "${meta.id}"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container 'quay.io/cellgeni/deepcell:0.12.10-0.0.2'

    input:
    tuple val(meta), val(x_min), val(y_min), val(x_max), val(y_max), path(image)

    output:
    tuple val(meta), path("${output_name}"), emit: wkts
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}_${x_min}_${y_min}_${x_max}_${y_max}"
    output_name = "${prefix}_deepcell_outlines.wkt"
    """
    deepcell_helper.py run \\
        ${image} \\
        -x_min ${x_min} \\
        -y_min ${y_min} \\
        -x_max ${x_max} \\
        -y_max ${y_max} \\
        -output_name ${output_name} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bioinfotongli: \$(deepcell_helper.py version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}_${x_min}_${y_min}_${x_max}_${y_max}"
    output_name = "${prefix}_deepcell_outlines.wkt"
    """
    touch ${output_name}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bioinfotongli: \$(deepcell_helper.py version)
    END_VERSIONS
    """
}
