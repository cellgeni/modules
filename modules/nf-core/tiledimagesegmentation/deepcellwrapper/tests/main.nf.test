nextflow_process {

    name "Test Process TILEDIMAGESEGMENTATION_DEEPCELLWRAPPER"
    script "../main.nf"
    process "TILEDIMAGESEGMENTATION_DEEPCELLWRAPPER"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "tiledimagesegmentation"
    tag "tiledimagesegmentation/deepcellwrapper"

    test("mouse_heart - tiff") {

        when {
            process {
                """
                input[0] = [
                    [ id:'mouse_heart'],
                    0, 0, 600, 600,
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("mouse_heart - tiff - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'mouse_heart'],
                    0, 0, 200, 200,
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}