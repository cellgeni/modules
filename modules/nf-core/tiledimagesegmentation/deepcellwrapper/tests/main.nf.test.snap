{
    "mouse_heart - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_0_0_200_200_deepcell_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,28dca11edc707da4d3253dda2ab465be"
                ],
                "versions": [
                    "versions.yml:md5,28dca11edc707da4d3253dda2ab465be"
                ],
                "wkts": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_0_0_200_200_deepcell_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-25T22:41:50.646530492"
    },
    "mouse_heart - tiff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_0_0_600_600_deepcell_outlines.wkt:md5,c9ed63452324e952cb3a51091738ba0b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,28dca11edc707da4d3253dda2ab465be"
                ],
                "versions": [
                    "versions.yml:md5,28dca11edc707da4d3253dda2ab465be"
                ],
                "wkts": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_0_0_600_600_deepcell_outlines.wkt:md5,c9ed63452324e952cb3a51091738ba0b"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-25T22:41:38.951922808"
    }
}